Adds Github Action for building on push to release branch (#4852)
knative/docs: contribute-to-docs/docs-release-process.md

1. PR the result to main.

## Update the dot-release Job

> Edit the `.github/workflows/knative-dot-release-build.yaml` file to the just-cut release. For example, if v1.2 was the version just cut:

Reviewer: ```suggestion
Edit the `.github/workflows/knative-dot-release-build.yaml` file to the just-cut release. For example, if v1.2 was the version just cut:
```
```suggestion
Edit the `.github/workflows/knative-dot-release-build.yaml` file to the new release. For example, if v1.2 was the version that you just cut:
```

> Edit the `.github/workflows/knative-dot-release-build.yaml` file to the just-cut release. For example, if v1.2 was the version just cut:

Reviewer: ```suggestion
Edit the `.github/workflows/knative-dot-release-build.yaml` file to the new release. For example, if you have just cut v1.2:
```

Reviewer: I'm also wondering if it would be useful to make `.github/workflows/knative-dot-release-build.yaml` link to the file in GitHub.

Reviewer: I'm also wondering if it would be useful to have `.github/workflows/knative-dot-release-build.yaml` link to the file in GitHub.

Author: :facepalm: yes, that would be good :smile:

> ## Update the dot-release Job

Reviewer: I just remembered, you probably need to add this as another step on line 12 above.
